libafl_qemu/src/qemu/qemu_config.rs

```diff
@@ -0,0 +1,342 @@
+use core::{
+    fmt,
+    fmt::{Display, Formatter},
+};
+use std::{
+    path::{Path, PathBuf},
+    sync::OnceLock,
+};
+
+use libafl_derive;
+use strum_macros;
+use typed_builder::TypedBuilder;
+
+use crate::{Qemu, QemuInitError};
+
+pub(super) static QEMU_CONFIG: OnceLock<QemuConfig> = OnceLock::new();
+
+#[cfg(emulation_mode = "systemmode")]
+#[allow(non_camel_case_types)]
+#[derive(Debug, strum_macros::Display, Clone)]
+#[strum(prefix = "-accel ")]
+pub enum Accelerator {
+    kvm,
+    tcg,
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, strum_macros::Display, Clone)]
+#[strum(prefix = "if=")]
+pub enum DriveInterface {
+    floppy,
+    ide,
+    mtd,
+    none,
+    pflash,
+    scsi,
+    sd,
+    virtio,
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, strum_macros::Display, Clone)]
+#[strum(prefix = "format=")]
+pub enum DiskImageFileFormat {
+    qcow2,
+    raw,
+}
+
+#[derive(Debug, Clone, Default, TypedBuilder)]
+pub struct Drive {
+    #[builder(default, setter(strip_option))]
+    file: Option<PathBuf>,
+    #[builder(default, setter(strip_option))]
+    format: Option<DiskImageFileFormat>,
+    #[builder(default, setter(strip_option))]
+    interface: Option<DriveInterface>,
+}
+
+impl Display for Drive {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "-drive")?;
+
+        let mut is_first_option = true;
+        let mut separator = || {
+            if is_first_option {
+                is_first_option = false;
+                " "
+            } else {
+                ","
+            }
+        };
+
+        if let Some(file) = &self.file {
+            write!(f, "{}file={}", separator(), file.to_str().unwrap())?;
+        }
+        if let Some(format) = &self.format {
+            write!(f, "{}{format}", separator())?;
+        }
+        if let Some(interface) = &self.interface {
+            write!(f, "{}{interface}", separator())?;
+        }
+
+        Ok(())
+    }
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, strum_macros::Display, Clone)]
+#[strum(prefix = "-serial ")]
+pub enum Serial {
+    none,
```

Comment:
How hard would it be to add some docs? Can we copy&paste then from qemu?

Reply:
I added a few lines where the names were not self documenting enough and I added a link to Qemu docs. Together with @rmalmain we thought that duplicating the docs could be less maintainable and maybe not too useful.
(In addition, in the long long long run this might be replaced with something auto-generated from QEMU haxe files)